adapters/zentotem/zentotem.go

```diff
@@ -0,0 +1,131 @@
+package zentotem
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the {bidder} adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if request == nil {
+		return nil, []error{&errortypes.BadInput{Message: "empty request"}}
+	}
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "empty request.Imp"}}
+	}
+
+	requests := make([]*adapters.RequestData, 0, len(request.Imp))
+	var errors []error
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		requestCopy.Imp = []openrtb2.Imp{imp}
+
+		requestJSON, err := json.Marshal(request)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    a.endpoint,
+			Body:   requestJSON,
+			ImpIDs: []string{imp.ID},
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errors
+}
+
+func getMediaTypeForBid(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
+	if imp.Native != nil {
+		return openrtb_ext.BidTypeNative, nil
+	}
+
+	if imp.Banner != nil {
+		return openrtb_ext.BidTypeBanner, nil
+	}
+
+	if imp.Video != nil {
+		return openrtb_ext.BidTypeVideo, nil
+	}
+
+	return "", &errortypes.BadInput{
+		Message: fmt.Sprintf("Processing an invalid impression; cannot resolve impression type for imp #%s", imp.ID),
+	}
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	impMap := map[string]openrtb2.Imp{}
+	for _, imp := range request.Imp {
+		impMap[imp.ID] = imp
+	}
```

Comment:
You should not be creating the impMap if you use Mtype field set in the bid response.

Reply:
fixed